datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -345,6 +444,55 @@ public StickyPartitionAssignmentStrategy(Optional<Integer> maxTasks, Optional<In
     return tasksToCleanUp;
   }
 
+  @Override
+  protected int constructExpectedNumberOfTasks(DatastreamGroup dg, List<String> instances,
+      Map<String, Set<DatastreamTask>> currentAssignmentCopy) {
+    // Count the number of tasks present for this datastream
+    Set<DatastreamTask> allAliveTasks = new HashSet<>();
+    for (String instance : instances) {
+      List<DatastreamTask> foundDatastreamTasks =
+          Optional.ofNullable(currentAssignmentCopy.get(instance)).map(c ->
+              c.stream().filter(x -> x.getTaskPrefix().equals(dg.getTaskPrefix()) && !allAliveTasks.contains(x))
+                  .collect(Collectors.toList())).orElse(Collections.emptyList());
+      allAliveTasks.addAll(foundDatastreamTasks);
+    }
+
+    boolean enableElasticTaskAssignment = getEnableElasticTaskAssignment(dg);
+    // TODO: Fetch the number of tasks in ZK if needed
+    int numTasks = enableElasticTaskAssignment ? _taskCountPerDatastreamGroup.getOrDefault(dg, 0) :
+        getNumTasks(dg, instances.size());
+
+    // Case 1: Elastic task assignment is disabled. Return getNumTasks() number of tasks.
```

Comment:
nit: plz fix this comment.

Reply:
done, I think. basically wanted to call out that if elastic task assignment is disabled, we set it to numTasks.